src/backend/marsha/deposit/serializers.py

```diff
@@ -0,0 +1,235 @@
+import mimetypes
+from os.path import splitext
+from urllib.parse import quote_plus
+
+from django.conf import settings
+from django.urls import reverse
+from django.utils import timezone
+
+from botocore.signers import CloudFrontSigner
+from rest_framework import serializers
+from rest_framework_simplejwt.models import TokenUser
+
+from marsha.core.serializers.playlist import PlaylistLiteSerializer
+from marsha.core.utils import cloudfront_utils, time_utils
+from marsha.core.utils.url_utils import build_absolute_uri_behind_proxy
+
+from ..core.serializers import (
+    InitiateUploadSerializer,
+    UploadableFileWithExtensionSerializerMixin,
+)
+from .models import DepositedFile, FileDepository
+
+
+class DepositedFileSerializer(
+    UploadableFileWithExtensionSerializerMixin, serializers.ModelSerializer
+):
+    """A serializer to display a deposited file."""
+
+    class Meta:  # noqa
+        model = DepositedFile
+        fields = (
+            "filename",
+            "author_name",
+            "id",
+            "file_depository",
+            "read",
+            "url",
+            "uploaded_on",
+            "upload_state",
+        )
+        read_only_fields = (
+            "id",
+            "file_depository",
+            "url",
+            "uploaded_on",
+            "upload_state",
+        )
+
+    # file_depository = FileDepositorySerializer(read_only=True)
+    # filename = serializers.SerializerMethodField()
+    url = serializers.SerializerMethodField()
+    # Make sure file depository UUID is converted to a string during serialization
+    file_depository = serializers.PrimaryKeyRelatedField(
+        read_only=True, pk_field=serializers.CharField()
+    )
+
+    def create(self, validated_data):
+        """Force the file depository field to the file depository of the JWT Token if any,
+        and set the author name to the username of the JWT Token if any.
+
+        Parameters
+        ----------
+        validated_data : dictionary
+            Dictionary of the deserialized values of each field after validation.
+
+        Returns
+        -------
+        dictionary
+            The "validated_data" dictionary is returned after modification.
+
+        """
+        # user here is a file depository as it comes from the JWT
+        # It is named "user" by convention in the `rest_framework_simplejwt` dependency we use.
+        user = self.context["request"].user
+        user_token = user.token.get("user")
+        # Set the file depository field from the payload if there is one and the user is identified
+        # as a proper user object through access rights
+        if (
+            self.initial_data.get("file_depository")
+            and user_token
+            and user.token["resource_id"] == user_token.get("id")
+        ):
+            validated_data["file_depository"] = self.initial_data.get("file_depository")
+            validated_data["author_name"] = user_token.get("username")
```

Comment:
According to previous comment, this can never happen, because user JWT doesn't carry this information, and the test above made sure this is not a resource JWT (which may have this information)